### ISSUE TITLE
Loading persisted cluster configuration state from disk does not work

### ISSUE THREAD

**Issue by simontherry** [dotnet/dotNext#105]
I think I found an issue with reading a changed configuration from disk.
Consider the following scenario:
- Start a few RAFT nodes which are not clustered
- Use AddMemberAsync to pair them and ApplyAsync to save it to disk
- Kill the nodes and restart them, the configuration is lost.

I was able to reproduce this in the ConfigurationRecovery test with the following code.

```cs
        [Fact]
        public static async Task ConfigurationRecovery()
        {
            var path = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
            var ep = new HttpEndPoint(new Uri("https://localhost:3262", UriKind.Absolute));
            var id = ClusterMemberId.FromEndPoint(ep);

            await using (var storage = new PersistentClusterConfigurationStorage(path))
            {
                await storage.LoadConfigurationAsync();
                True(await storage.AddMemberAsync(id, ep));
                await storage.ApplyAsync();
            }

            var ep2 = new HttpEndPoint(new Uri("https://localhost:3263", UriKind.Absolute));
            var id2 = ClusterMemberId.FromEndPoint(ep2);

            await using (var storage = new PersistentClusterConfigurationStorage(path))
            {
                await storage.LoadConfigurationAsync();
                True(await storage.AddMemberAsync(id2, ep2));
                await storage.ApplyAsync();
            }

            await using (var storage = new PersistentClusterConfigurationStorage(path))
            {
                await storage.LoadConfigurationAsync();
                Null(storage.As<IClusterConfigurationStorage<HttpEndPoint>>().ProposedConfiguration);
                Equal(ep, storage.As<IClusterConfigurationStorage<HttpEndPoint>>().ActiveConfiguration[id]);
                Equal(ep2, storage.As<IClusterConfigurationStorage<HttpEndPoint>>().ActiveConfiguration[id2]);
            }
        }
```

Culprit seems to be in PersistentClusterConfigurationStorage.cs where I had to modify the code to also execute 
```cs
output.fs.Position = 0L;
```

so we get this code:

```cs
        internal async ValueTask CopyToAsync(ClusterConfiguration output, int bufferSize, CancellationToken token)
        {
            output.fs.SetLength(Length);
            output.Fingerprint = Fingerprint;
            fs.Position = 0L;
            output.fs.Position = 0L;
            await fs.CopyToAsync(output.fs, bufferSize, token).ConfigureAwait(false);
            await output.fs.FlushAsync(token).ConfigureAwait(false);
        }
```

Can you confirm this is indeed the correct approach and is indeed a bug? 

Thanks!

**Comment by sakno** [comment]
@simontherry , thanks for this finding. It seems like a bug. Would you like to make a PR with fix and tests?

**Comment by simontherry** [comment]
Added pull request #106

**Comment by sakno** [comment]
Fixed in release 4.4.1
